pkg/ccl/streamingccl/streamproducer/stream_lifetime.go

```diff
@@ -307,3 +326,37 @@ func completeReplicationStream(
 		return nil
 	})
 }
+
+func setupSpanConfigsStream(
+	ctx context.Context, eval *eval.Context, txn isql.Txn, tenantName roachpb.TenantName,
+) (*streampb.ReplicationStreamSpec, error) {
+
+	tenantRecord, err := sql.GetTenantRecordByName(ctx, eval.Settings, txn, tenantName)
+	if err != nil {
+		return nil, err
+	}
+
+	tenantID := roachpb.MustMakeTenantID(tenantRecord.ID)
+
+	var spanConfigID descpb.ID
+	execConfig := eval.Planner.ExecutorConfig().(*sql.ExecutorConfig)
+	if err := sql.DescsTxn(ctx, execConfig, func(ctx context.Context, txn isql.Txn, col *descs.Collection) error {
```

Comment:
Does this run in the transaction we are already a part of? Should it?

Reply:
> Does this run in the transaction we are already a part of?

This sql.DescsTxn() creates creates a new txn via the executorConfig, which I believe is independent of the `p.InternalSQLTxn()` that we [initialize ](https://github.com/cockroachdb/cockroach/blob/574a09e39025ae5021d9d224e061168fd01aeda3/pkg/sql/planner.go#L910)the replication manager with. 


> Should it?

I'm not totally sure. If the replication manager is _supposed_ to read from kv only via the `replicationManager.Txn`, part of the [replication manager state](https://github.com/cockroachdb/cockroach/blob/574a09e39025ae5021d9d224e061168fd01aeda3/pkg/ccl/streamingccl/streamproducer/replication_manager.go#L30), then we're being naughty:
1. https://github.com/cockroachdb/cockroach/blob/574a09e39025ae5021d9d224e061168fd01aeda3/pkg/ccl/streamingccl/streamproducer/stream_lifetime.go#L187
2. https://github.com/cockroachdb/cockroach/blob/574a09e39025ae5021d9d224e061168fd01aeda3/pkg/ccl/streamingccl/streamproducer/stream_lifetime.go#L222

My hunch is that we probably ought to use the same txn for the examples I list and in this PR, as its cleanest to read the db from one consistent timestamp. Do you agree? 

(FWIW, I was questioning why we even had this seperate stateful txn object, and [here's](https://github.com/cockroachdb/cockroach/pull/90964) why. )